tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AdoNet/DbScopeFactory.cs

```diff
@@ -28,14 +28,16 @@ private static Scope CreateDbCommandScope(Tracer tracer, IDbCommand command, Int
             }
 
             Scope scope = null;
+            string commandText = command.CommandText;
 
             try
             {
                 Span parent = tracer.InternalActiveScope?.Span;
 
                 if (parent is { Type: SpanTypes.Sql } &&
                     parent.GetTag(Tags.DbType) == dbType &&
-                    parent.ResourceName == command.CommandText)
+                    (parent.GetTag(Tags.DbmDataPropagated) != null ||
+                    parent.ResourceName == commandText))
```

Comment:
Two really small notes:
1. If the default value of this feature for now is `disabled`, then we should put the ResourceName comparison first so we exit sooner
2. For readability, it might help to put both parts of the `||` comparison on the same line